### PR TITLE
simple intro slides

### PR DESCRIPTION
**Pull request by eparejatobes**

### CODE REVIEW

**Comment by laughedelic** [comment]
could you just copy everything from [here](https://github.com/ohnosequences/puntos-y-flechicas/tree/master/slides/ugr) and change the slides?

File: docs/slides/slides.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+## Design
+
+- (kinda) contains Java
+- functional features
+- strict eval by default
+- funny type system
+
+### wait, how?
+
+is complicated...
```
Comment (laughedelic) [inline docs/slides/slides.md]: isn't espanish :stuck_out_tongue:

File: docs/slides/slides.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+val zu = z + u.size
+// nonono
+zu = zu + 3
+```
+
+### Recursion support
+
+``` scala
+import scala.annotation.tailrec
+@tailrec def factorial_rec(x: Int, y: Int): Int =  if (x == 1) y else factorial_rec(x - 1, x * y)
```
Comment (laughedelic) [inline docs/slides/slides.md:88]: one line!!! :ok_hand:

File: docs/slides/slides.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+### Type abstraction
+
+``` scala
+type L[Z] = List[Z]
+trait Function {
+
+  type Domain
+  type Codomain
+  def apply(x: Domain): Codomain
+}
```
Comment (laughedelic) [inline docs/slides/slides.md:231]: Some useful example of how to use such `Function`?

File: docs/slides/slides.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+```
+
+### Type abstraction II
+
+``` scala
+class list[X]
+trait A
+trait C[Z <: A]
```
Comment (laughedelic) [inline docs/slides/slides.md]: What's this about?

File: docs/slides/slides.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+### DSLs
+
+- implicits
+- overloading
+- minimal set of reserved keywords
+
+<!-- TODO explain, why Scala helps, examples? -->
+
+- https://github.com/fogus/baysick !!!
```
Comment (laughedelic) [inline docs/slides/slides.md]: :+1: :+1: :+1:

File: docs/slides/slides.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+### Ecosystem
+
+- Spark
+- Akka ??
+- Twitter stuff (Finagle and the like)
+- https://github.com/non/spire
```
Comment (laughedelic) [inline docs/slides/slides.md]: shapeless? :wink: or a separate slide with some freaky stuff...

File: docs/slides/slides.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+## Session structure
+
+1. Generalities blahblahblah
+2. show me teh codez!
+
+## Infrastructure
+
+- **JVM**-based (GC, optimizations, ...)
+- **Java** interoperability
+- (can be) as fast as **Java**, sometimes more!
```
Comment (laughedelic) [inline docs/slides/slides.md:21]: O RLY?

File: docs/slides/slides.md (hunk 1 above)
Comment (eparejatobes) [inline docs/slides/slides.md]: no me sale en español... :-|

File: docs/slides/slides.md (hunk 2 above)
Comment (eparejatobes) [inline docs/slides/slides.md:88]: one stack!

File: docs/slides/slides.md (hunk 3 above)
Comment (eparejatobes) [inline docs/slides/slides.md:231]: you can compose them :P

File: docs/slides/slides.md (hunk 6 above)
Comment (eparejatobes) [inline docs/slides/slides.md]: I can add some of my homegrown categorical programming code

File: docs/slides/slides.md (hunk 7 above)
Comment (eparejatobes) [inline docs/slides/slides.md:21]: yezz run teh codezzz

**Comment by eparejatobes** [comment]
Reviewing this with @laughedelic
1. typeclasses example with equals. Explain it as adding methods for Java people?
2. something more about objects?
3. no refinements, later in the code part
4. simple ADTs (pattern matching)

**Pull request merged**